apps/api/src/modules/preset/handlers/preset.service.ts

```diff
@@ -0,0 +1,73 @@
+import {HttpErrors} from '@fastify/sensible/lib/httpError';
+import {PresetCreateRequest, PresetCreateResponse} from '../domain';
+import {PresetRepository} from '../repository';
+type PresetCompleteResponse = PresetCreateResponse & {isOwner: boolean};
+export interface PresetService {
+  findById(ownerId: string | null, id: string): Promise<PresetCompleteResponse>;
+  create(
+    userId: string,
+    data: PresetCreateRequest,
+  ): Promise<PresetCreateResponse>;
+  update: (
+    userId: string,
+    id: string,
+    data: PresetCreateRequest,
+  ) => Promise<PresetCreateResponse>;
+  deletePreset: (userId: string, id: string) => Promise<PresetCreateResponse>;
+}
+export const makePresetService = (
+  repository: PresetRepository,
+  httpErrors: HttpErrors,
+): PresetService => {
+  const findById = async (
+    ownerId: string,
+    id: string,
+  ): Promise<PresetCompleteResponse> => {
+    const preset = await repository.findById(id);
+    if (!preset) {
+      throw httpErrors.notFound(`Preset with id ${id} not found`);
+    }
+    const isOwner = !!ownerId && ownerId === preset.ownerId;
+
+    return {...preset, isOwner};
```

Comment:
This part could not be necessary. Since presets are only for user scope at the moment, we can do a findByIdAndOwnerId directly.

Then if the user that's requesting the resource is different from the owner we will throw a not found exception